core/src/main/java/com/google/errorprone/bugpatterns/EmptyCatch.java

```diff
@@ -0,0 +1,389 @@
+import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
+
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker.TryTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.LineMap;
+
+import static com.sun.source.tree.Tree.Kind;
+import static com.sun.source.tree.Tree.Kind.RETURN;
+import static com.sun.source.tree.Tree.Kind.CONTINUE;
+import static com.sun.source.tree.Tree.Kind.BREAK;
+import static com.sun.source.tree.Tree.Kind.ASSIGNMENT;
+import static com.sun.source.tree.Tree.Kind.EXPRESSION_STATEMENT;
+import static com.sun.source.tree.Tree.Kind.EMPTY_STATEMENT;
+import static com.sun.source.tree.Tree.Kind.CATCH;
+import static com.sun.source.tree.Tree.Kind.BLOCK;
+import static com.sun.source.tree.Tree.Kind.IF;
+import static com.sun.source.tree.Tree.Kind.FOR_LOOP;
+import static com.sun.source.tree.Tree.Kind.DO_WHILE_LOOP;
+import static com.sun.source.tree.Tree.Kind.WHILE_LOOP;
+import static com.sun.source.tree.Tree.Kind.ENHANCED_FOR_LOOP;
+import static com.sun.source.tree.Tree.Kind.SWITCH;
+import static com.sun.source.tree.Tree.Kind.TRY;
+import static com.sun.source.tree.Tree.Kind.VARIABLE;
+
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.CatchTree;
+import com.sun.source.tree.StatementTree;
+import com.sun.source.tree.TryTree;
+import com.sun.source.tree.BlockTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.ExpressionStatementTree;
+
+import com.sun.source.util.TreePath;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.tree.JCTree;
+import com.sun.tools.javac.tree.TreeInfo;
+
+import java.util.List;
+
+/**
+ * A bug checker for empty exception handling block:
+ *
+ * <pre>
+ * try {
+ *   // do something
+ * } catch (XX e) {
+ *   // empty, only comment, or only log printing statements
+ * }
+ * </pre>
+ * We also implement a variety of false positive suppression heuristics.
+ * 
+ * For more detail, refer to the paper:
+ * "Simple Testing Can Prevent Most Critical Failures: 
+ *  An Analysis of Production Failures in Distributed Data-intensive Systems"
+ *  Yuan et al. Proceedings of the 11th USENIX Symposium on Operating Systems Design 
+ *  and Implementation (OSDI), 2014
+ *
+ * @author yuan@eecg.utoronto.ca (Ding Yuan)
+ */
+@BugPattern(name = "EmptyCatch",
+    summary = "Empty catch blocks that ignore exceptions",
+    explanation = "Ignoring exception is a bad practice.",
+    category = JDK, maturity = MATURE, severity = WARNING)
+public class EmptyCatch extends BugChecker implements TryTreeMatcher {
+  /* If one of the following exceptions is ignored, we suppress the warning. */
+  String[] harmlessExceptions = {
```

Comment:
Two things:
1. I would use the fully qualified types here, e.g. "java.io.FileNotFoundException".
2. Please use a set rather than a String array: `ImmutableSet<String> harmlessExceptions = ImmutableSet.of("java.io.FileNotFoundException", "java.lang.ReflectiveOperationException", ...);`
Then you can do the membership check by just doing `harmlessExceptions.contains(typeString);`